packages/@o3r/extractors/src/core/comparator/package-managers-extractors/yarn2-file-extractor.helper.ts

```diff
@@ -0,0 +1,72 @@
+import { Cache, Configuration, Descriptor, FetchOptions, FetchResult, Locator, Project, ResolveOptions, structUtils, ThrowReport } from '@yarnpkg/core';
+import { npath, PortablePath } from '@yarnpkg/fslib';
+import * as npmPlugin from '@yarnpkg/plugin-npm';
+import { join } from 'node:path';
+
+// TODO : how can we handle the yarn version discrepancy between user and package v?
+// Test with different yarn major versions configured for wrks
+async function getProject(): Promise<Project> {
+  const pluginConfiguration = {
+    plugins: new Set(['plugin-npm']),
+    modules: new Map([['plugin-npm', npmPlugin.default]])
+  };
+  const configuration = await Configuration.find(npath.toPortablePath(process.cwd()), pluginConfiguration, { strict: false });
+  if (!configuration.projectCwd) {
+    throw new Error(`No project found from ${process.cwd()}`);
+  }
+  const { project } = await Project.find(configuration, configuration.projectCwd);
+  return project;
+}
+
+function getDescriptorFromReference(packageReference: string): Descriptor {
+  const descriptor = structUtils.tryParseDescriptor(packageReference);
+  if (!descriptor) {
+    throw new Error(`Invalid package descriptor ${packageReference}`);
+  }
+  return descriptor;
+}
+
+async function fetchPackage(project: Project, descriptor: Descriptor): Promise<FetchResult> {
+  const cache = await Cache.find(project.configuration);
+  const report = new ThrowReport();
+  const resolver = project.configuration.makeResolver();
+  const fetcher = project.configuration.makeFetcher();
+  const fetchOptions: FetchOptions = { project, cache, checksums: project.storedChecksums, report, fetcher };
+  const resolveOptions: ResolveOptions = { project, report, resolver, fetchOptions };
+
+  const normalizedDescriptor = project.configuration.normalizeDependency(descriptor);
+  const candidate = await resolver.getCandidates(normalizedDescriptor, {}, resolveOptions);
+
+  if (candidate.length === 0) {
+    throw new Error(`No candidate found for @${descriptor.scope}/${descriptor.name}@${descriptor.range}`);
+  }
+  const isSupported = fetcher.supports(candidate[0], fetchOptions);
+  if (!isSupported) {
+    throw new Error(`Fetcher does not support @${descriptor.scope}/${descriptor.name}@${descriptor.range}`);
+  }
+  return fetcher.fetch(candidate[0], resolveOptions.fetchOptions!);
+}
+
+function getRelativePathInCache(locator: Locator | Descriptor, path: string): PortablePath {
+  const cachePath = join('./node_modules', locator.scope ? '@' + locator.scope : '', locator.name, path);
+  return npath.toPortablePath(cachePath);
+}
+
+/**
+ * Retrieves the list of given files from an npm package using yarn.
```

Comment:
Maybe you should specify that this is for a package not part of the dependencies that you download as well.
Because according to the description it seems to a simple `fs.readFile(require.resolve(path))`